### PR TITLE
[ImagePicker] Captured Video quality and duration support

### PR DESCRIPTION
**Pull request by 0xcodezero**
## Description

*Starting using the Plugin, I found that there are no extra parameters could be passed to the pickVideo function. This behavior limits the usage of the library especially having no record duration could lead to a huge file size that is very hard to upload to a server.

In this PR, I have implemented optional two parameters for pickVideo function (quality, durationInSeconds) which were covered in the three levels (Dart, iOS Objective-C, and Android Java Library) that controlled the outcome video recorded from the device Camera

The changes are on iOS (ImagePickerPlugin.m), Android(ImagePickerDelegate.java), Dart (image_picker.dart),(test/image_picker_test.dart).

Being implemented as optional parameters, it will not break any previous code implementations and supports the default behavior as currently implemented if the parameters are not passed. (High-Quality video, No duration record duration).
*

## Related Issues

*[flutter/flutter#40035](https://github.com/flutter/flutter/issues/40035)*

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by cyanglaz** [comment]
The video duration has been worked on https://github.com/flutter/plugins/pull/2048 and it is close to be landed.
@0xcodezero Do you want to modify your PR to only add quality?

File: packages/image_picker/lib/image_picker.dart [hunk 1]
```diff
@@ -23,6 +23,18 @@ enum ImageSource {
   gallery,
 }
 
+/// Specifies the video quality to be picked from Camera.
+enum VideoQuality {
+  /// High Qulaity Video with Maximum size.
```
Comment (ened) [inline packages/image_picker/lib/image_picker.dart]: "Quality"

File: packages/image_picker/lib/image_picker.dart [hunk 2]
```diff
@@ -23,6 +23,18 @@ enum ImageSource {
...
+  High,
+
+  /// Medium Qulaity Video.
```
Comment (ened) [inline packages/image_picker/lib/image_picker.dart]: "Quality"

File: packages/image_picker/lib/image_picker.dart [hunk 3]
```diff
@@ -23,6 +23,18 @@ enum ImageSource {
...
+  Medium,
+
+  /// Low Qulaity Video.
```
Comment (ened) [inline packages/image_picker/lib/image_picker.dart]: "Quality"

File: packages/image_picker/lib/image_picker.dart [hunk 4]
```diff
@@ -79,16 +91,31 @@ class ImagePicker {
   /// The [source] argument controls where the video comes from. This can
   /// be either [ImageSource.camera] or [ImageSource.gallery].
   ///
+  /// The [quality] optional argument controls the video quality and its size on correspondence. This can
```
Comment (ened) [inline packages/image_picker/lib/image_picker.dart]: @cyanglaz should we enforce 80 char line length?

File: packages/image_picker/lib/image_picker.dart (hunk 3 above)
Comment (0xcodezero) [reply packages/image_picker/lib/image_picker.dart]: Done for all three typos, thanks @ened

File: packages/image_picker/lib/image_picker.dart (hunk 4 above)
Comment (0xcodezero) [reply packages/image_picker/lib/image_picker.dart]: I break it into two Lines, as the default value was originally in a separate statement.

**Comment by 0xcodezero** [comment]
> The video duration has been worked on #2048 and it is close to be landed.
> @0xcodezero Do you want to modify your PR to only add quality?

@cyanglaz I will wait until his PR merged into master, and I will update mine, and resolve any conflicts if required. is it Ok?

**Comment by cyanglaz** [comment]
@0xcodezero FYI the video duration is landed in flutter/plugins#2643. You can rebase and focus this PR on quality now.

**Comment by fedeandreocci** [comment]
@cyanglaz hi! Is there a possibility that this PR will be merged? quality parameter for recorded video would be a stunning feature!

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**